File: web/src/api/personality.ts
import api from './client'

export interface Personality {
  id: string
  name: string
  description: string
  system_prompt: string
  is_active: boolean
  traits: PersonalityTrait[]
  created_at: string
  updated_at: string
}

export interface PersonalityTrait {
  key: string
  value: string
  weight: number
}

export const personalityApi = {
  list: () => api.get<Personality[]>('/personalities'),

  get: (id: string) => api.get<Personality>(`/personalities/${id}`),

  create: (data: { name: string; description: string; system_prompt: string }) =>
    api.post<Personality>('/personalities', data),

  update: (id: string, data: { name: string; description: string; system_prompt: string }) =>
    api.put<Personality>(`/personalities/${id}`, data),

  delete: (id: string) => api.delete<{ message: string }>(`/personalities/${id}`),

  activate: (id: string) => api.post<{ message: string }>(`/personalities/${id}/activate`),

  getActive: () => api.get<Personality>('/personalities/active'),
}

File: server/internal/personality/model/personality.go
package model

import (
	"fmt"
	"time"

	"github.com/google/uuid"
	"github.com/IceWhaleTech/ZimaOS-Blue/server/internal/timeutil"
)

// Personality represents an AI assistant personality
type Personality struct {
	ID           string           `json:"id"`
	Name         string           `json:"name"`
	Description  string           `json:"description"`
	SystemPrompt string           `json:"system_prompt"`
	IsActive     bool             `json:"is_active"`
	Traits       []PersonalityTrait `json:"traits"`
	CreatedAt    time.Time        `json:"created_at"`
	UpdatedAt    time.Time        `json:"updated_at"`
}

// PersonalityTrait represents a personality characteristic
type PersonalityTrait struct {
	Key    string  `json:"key"`
	Value  string  `json:"value"`
	Weight float64 `json:"weight"`
}

// NewPersonality creates a new personality
func NewPersonality(name, description, systemPrompt string) *Personality {
	return &Personality{
		ID:           uuid.New().String(),
		Name:         name,
		Description:  description,
		SystemPrompt: systemPrompt,
		Traits:       []PersonalityTrait{},
		CreatedAt:    timeutil.NowTime(),
		UpdatedAt:    timeutil.NowTime(),
	}
}

// AddTrait adds a trait to the personality
func (p *Personality) AddTrait(key, value string, weight float64) {
	p.Traits = append(p.Traits, PersonalityTrait{
		Key:    key,
		Value:  value,
		Weight: weight,
	})
	p.UpdatedAt = timeutil.NowTime()
}

// Validate validates the personality
func (p *Personality) Validate() error {
	if p.Name == "" {
		return fmt.Errorf("personality name cannot be empty")
	}
	if p.SystemPrompt == "" {
		return fmt.Errorf("system prompt cannot be empty")
	}
	return nil
}
